Assets/UIElements/Scripts/XRUI_2DDragInteractable.cs

```diff
@@ -77,8 +77,13 @@ private void UpdateByMovement()
         float posX = transform.localPosition.x;
         posX = posX > 1 ? 1 : posX;
         posX = posX < 0 ? 0 : posX;
+        if (posX <= 0 || posX >= 1)
+        {
+            this.GetComponent<Rigidbody>().velocity = Vector3.zero;
+            this.GetComponent<Rigidbody>().angularVelocity = Vector3.zero;
+        }
```

Comment:
Why this if is created here if the ternary conditions above forcing the posX to be between 0 - 1?

Reply:
Removed because we set iskinematic to true and its will avoid some bugs